### PR TITLE
feat: Add Snap UI Radio component

### PR DESCRIPTION
**Pull request by david0xd**
## **Description**
This PR integrates new Radio components for Snap UI.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26181?quickstart=1)

## **Related issues**
Fixes: https://github.com/MetaMask/MetaMask-planning/issues/1571

## **Manual testing steps**
1. Go to `Test Snaps` and install `Interactive UI Snap`.
2. Run Interactive UI Snap by creating a dialog.
3. Check that radio options are available in the UI and confirm their functionality. (Latest Test Snaps needs to be released before this is possible, use `main` branch in Snaps repo and run locally if necessary)

### **Before**
Nothing to show. Radio options were not available before.

### **After**
Screenshot 2024-07-29 at 11 29 00
Screenshot 2024-07-29 at 11 29 17
Screenshot 2024-07-29 at 11 29 40

## **Pre-merge author checklist**
- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/app/snaps/snap-ui-radio-group/snap-ui-radio-group.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  form,
+  ...props
+}) => {
+  const { handleInputChange, getValue } = useSnapInterfaceContext();
+
+  const initialValue = getValue<string>(name, form);
+
+  const [value, setValue] = useState(initialValue ?? '');
+
+  type RadioOptions = [{ value: string; name: string }];
```
Comment (ritave) [inline ui/components/app/snaps/snap-ui-radio-group/snap-ui-radio-group.tsx:38]: This is a tuple of only one element, this doesn't look necessary.

I suggest moving RadioOptions outside the function and using it in props.

```typescript
export type SnapUIRadioOption = {value: string, name: string};

export type SnapUIRadioProps = {
  options: SnapUIRadioOption[]
}

export const SnapUIRadioGroup: FunctionComponent<SnapUIRadioProps> = ({
  name,
  label,
  error,
  form,
  ...props
}) => {
  const displayRadioOptions = (options: SnapUIRadioOption[]) => {
    options.map((options) => {}); // this should work automatically
    // but if type is necessary in lambda
    options.map((options: SnapUIRadioOption) => {});
  }
}
```

**Review by ritave — CHANGES_REQUESTED** [review]

File: ui/components/app/snaps/snap-ui-radio-group/snap-ui-radio-group.tsx (hunk 1 above)
Comment (david0xd) [reply ui/components/app/snaps/snap-ui-radio-group/snap-ui-radio-group.tsx]: Refactored.

File: ui/components/app/snaps/snap-ui-radio-group/snap-ui-radio-group.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+  useEffect(() => {
+    if (initialValue) {
```
[the remark this answers is not in the document]
Comment (david0xd) [reply ui/components/app/snaps/snap-ui-radio-group/snap-ui-radio-group.tsx]: Updated.

**Review by ritave — APPROVED** [review]

File: ui/components/app/snaps/snap-ui-renderer/components/index.ts [hunk 3]
```diff
@@ -40,6 +41,7 @@ export const COMPONENT_MAPPING = {
   Link: link,
   Field: field,
   Dropdown: dropdown,
+  Radio: radioGroup,
```
Comment (FrederikBolding) [inline ui/components/app/snaps/snap-ui-renderer/components/index.ts:44]:
```suggestion
  RadioGroup: radioGroup,
```

File: ui/components/app/snaps/snap-ui-radio-group/snap-ui-radio-group.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+export const SnapUIRadioGroup: FunctionComponent<SnapUIRadioProps> = ({
+  name,
+  label,
+  error,
+  form,
+  ...props
+}) => {
+  const { handleInputChange, getValue } = useSnapInterfaceContext();
+
+  const initialValue = getValue<string>(name, form);
```
Comment (FrederikBolding) [inline ui/components/app/snaps/snap-ui-radio-group/snap-ui-radio-group.tsx:36]:
```suggestion
  const initialValue = getValue(name, form) as string;
```

I changed the typing for this in the base branch

File: ui/components/app/snaps/snap-ui-radio-group/snap-ui-radio-group.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+  Box,
+  HelpText,
+  HelpTextSeverity,
+  Label,
+  Text,
+} from '../../../component-library';
+
+export type SnapUIRadioOption = { value: string; name: string };
+
+export type SnapUIRadioProps = {
```
Comment (FrederikBolding) [inline ui/components/app/snaps/snap-ui-radio-group/snap-ui-radio-group.tsx:19]:
```suggestion
export type SnapUIRadioGroupProps = {
```

File: ui/components/app/snaps/snap-ui-radio-group/snap-ui-radio-group.tsx
Comment (FrederikBolding) [inline ui/components/app/snaps/snap-ui-radio-group/snap-ui-radio-group.tsx:1]: Please exclude `Radio` here:
https://github.com/MetaMask/metamask-extension/blob/dd/snap-radio-ui-component/ui/components/app/snaps/snap-ui-renderer/snap-ui-renderer.test.js#L4

https://github.com/MetaMask/metamask-extension/blob/dd/snap-radio-ui-component/ui/components/app/snaps/snap-ui-renderer/utils.ts#L92

**Review by FrederikBolding — APPROVED** [review]

File: ui/components/app/snaps/snap-ui-radio-group/snap-ui-radio-group.tsx (hunk 4 above)
Comment (david0xd) [reply ui/components/app/snaps/snap-ui-radio-group/snap-ui-radio-group.tsx]: Updated.

File: ui/components/app/snaps/snap-ui-radio-group/snap-ui-radio-group.tsx (hunk 5 above)
Comment (david0xd) [reply ui/components/app/snaps/snap-ui-radio-group/snap-ui-radio-group.tsx]: Updated.

File: ui/components/app/snaps/snap-ui-renderer/components/index.ts (hunk 3 above)
Comment (david0xd) [reply ui/components/app/snaps/snap-ui-renderer/components/index.ts]: Updated.

**Pull request merged**